app/models/company.rb

```diff
@@ -1,6 +1,6 @@
 class Company < ActiveRecord::Base
-  has_many :users
-  has_many :notes
+  has_many :users, dependent: :destroy
```

Comment:
This makes it so that, if a company is deleted from the db, it deletes its associated users.